src/Payments.sol

```diff
@@ -234,36 +315,47 @@ contract Payments is
         noRailModificationInProgress(railId)
     {
         Rail storage rail = rails[railId];
+
+        // operator can only reduce fixed lockup on a terminated rail
+        // it cannot change the lockup period or increase fixed lockup
+        require(
+            rail.terminationEpoch == 0 || (period == rail.lockupPeriod && lockupFixed <= rail.lockupFixed),
+            "failed to modify terminated rail: cannot change period or increase fixed lockup"
+        );
+
         Account storage payer = accounts[rail.token][rail.from];
         OperatorApproval storage approval = operatorApprovals[rail.token][
             rail.from
         ][rail.operator];
 
-        require(approval.isApproved, "operator not approved");
-
         // settle account lockup and if account lockup is not settled upto to the current epoch; revert
-        (bool fullySettled, uint256 lockupSettledUpto) = settleAccountLockup(
-            payer
-        );
+        uint256 lockupSettledUpto = settleAccountLockup(payer);
         require(
-            fullySettled && lockupSettledUpto == block.number,
+            lockupSettledUpto == block.number,
```

Comment:
note: technically it should be fine if _either_ the rail is terminated or lockup has been settled. Terminated rails don't lock any additional funds.

(we can leave this for now, but we may want to change it in the future)

Reply:
Done. You are right.